src/components/TablePanel.tsx

```diff
@@ -30,7 +32,30 @@ const TablePanel: FunctionComponent<TabelPanelProps> = ({
     updateTables();
   }, [connection, results]);
 
-  const links = Object.keys(tables).map((t) => ({
+  useEffect(() => {
+
+    const cols:string[] = []
+    const grps:IGroup[] = []
+
+    Object.entries(tables).forEach(([k,v]) => {      
```

Comment:
Rather than using `forEach` and pushing into these two arrays, how about ...

~~~
const cols =  Object.entries(tables).map([_, v] => v);
const grps = Object.entries(tables).( ...)
~~~

Reply:
That feels like that would be less performant as it's doing 2 loops instead of 1?